### ISSUE TITLE
Differences in permissions in package installation

### ISSUE THREAD

**Issue by krzyslom** [rstudio/renv#331]
Hi,

I'm running a docker container strictly based on rocker/verse:3.6.1 image.
If i exec into the container as root and run `R --slave -e "install.packages('here')"` I'll get the following result for `ls -l /usr/local/lib/R/site-library | grep here`
drwxrwxr-x 6 root staff ... here
However if I run `R --slave -e "renv::install('here')"` then I'll get
drwxr-xr-x 6 root staff ... here
This results in not being able to reinstall the package from the RStudio session while being rstudio user.

Strangely if I change the user from root to rstudio with `su rstudio` then both of the mentioned installation commands return the same result
drwxrwxr-x 6 rstudio staff ... here

To sum it up, somehow there is a difference in package folder permissions between `install.packages` and `renv::install` only while being the root user. This has a major impact during the Docker image build. I'm aware that I could mount the cache to the Docker container, however there are some functionalities and packages I would like to have already embedded inside the image.

The simple workaround could be to just run extra chmod command during the build, after the package installation.

However there must be a reason why this difference exists and maybe you would like to investigate it further.

**Comment by kevinushey** [comment]
I believe this is a consequence of `renv`'s staged install. `renv` will first install packages into a temporary library path, and then (if that install is successful) move that to the destination library path. That directory gets created with "regular" permissions, and then moved over.

R tries to respect the library's group writable permission bit here:

https://github.com/wch/r-source/blob/3a1693cabd68ae8a8003ff2b74d0ba20b51a092e/src/library/tools/R/install.R#L2143-L2147

I believe `renv` needs to do a similar permissions fix-up following a staged install.

**Comment by kevinushey** [comment]
The other solution here would be to give the `rstudio` user its own user library. However, `rocker` images are set up so that `R_LIBS_USER` actually points to the site library. This is a bit weird to me but I guess it must be intentional for some reason?

**Comment by kevinushey** [comment]
One other thing to consider here is how this plays with the cache, since permissions (on Linux) apply to the linked-to file, rather than the symlink itself.

**Comment by kevinushey** [comment]
I believe this should be resolved by https://github.com/rstudio/renv/commit/bc260019481ae56d62395abb7c606be1fd2c44f9 -- we now try to ensure the staging library has the same permissions as the target library.
